### PR TITLE
Added PKCS12 parsing.

### PR DESCRIPTION
**Pull request by idyll**
I moved to a hash for the certificate option so that I a file or a string could be used.
We pretty much never use the certificate option as a file and I wanted to make that option a first class citizen instead of the string io hack.

Certificate now supports the PKCS12 format as part of the option hash.

PKCS12 is the binary format you get from the Keychain.app.

### CODE REVIEW

**Comment by idyll** [comment]
I'm open to suggestions other that the hash, but I wanted to avoid forcing something onto the filesystem or into a string io.

The P12 option should really help people that are just getting push working for the first time. I've always found that manual conversion to PEM strange, especially when Ruby can just read the P12.

File: README.md [hunk 1]
```diff
@@ -49,11 +49,21 @@ pusher = Grocer.pusher(
...
+  certificate: { pem_file: "/path/to/cert.pem" }
+```
+
+
 #### Notes
 
-* `certificate`: If you don't have the certificate stored in a file, you
-  can pass any object that responds to `read`.
-  Example: `certificate: StringIO.new(pem_string)`
+* `certificate`: the old file/string io option is still supported but now certificate
```
Comment (stevenharman) [inline README.md]: We'd still want the documentation here for "the old file/string IO option."

File: lib/grocer/ssl_connection.rb [hunk 2]
```diff
@@ -20,11 +20,22 @@ def connected?
...
     end
 
-    def connect
-      context = OpenSSL::SSL::SSLContext.new
-
-      if certificate
+    def key_and_cert_from_options
+      case certificate
+      when Hash
+        if certificate['p12'] || certificate[:p12] || certificate['p12_file'] || certificate[:p12_file]
```
Comment (stevenharman) [inline lib/grocer/ssl_connection.rb]: We have a `DeepSymbolizeKeys` module which might clean this up?

**Comment by stevenharman** [comment]
Thanks for the thought and effort of this PR. Thinking down the road, I'm hesitant to add too many configuration options and wonder if we could get by with just allowing you to specify `certificate: { p12: foo }` or `certificate: { pem: foo }`. We then make the assumption that the value of the `:p12` or `:pem` key is also either a path to a file or it responds to `#read`.

Thoughts?

**Comment by idyll** [comment]
Heh. I was trying to avoid read, it feels awkward to me because I don't think grocer should have to deal with missing files or I/O that can't be read. I only left it in for legacy reasons. I would prefer to pass the contents of the file in the config block as opposed to the file itself.

I also have been trying to support a case where grocer is used with several connections to APNS managed by the https://github.com/mperham/connection_pool gem.

**Pull request closed without merging**